src/app/selectors.js

```diff
@@ -367,9 +370,35 @@ const countModelStatusGroups = (groupedModelStatuses) => {
   @param {Object} modelData The model data from the redux store.
   @returns {Object} The filtered model data.
 */
-const filterModelData = (filters, modelData) => {
+const filterModelData = (filters, modelData, controllerData) => {
+  const clonedModelData = cloneDeep(modelData);
+  // Add controllerName to the model data by mapping the controllerData
+  Object.entries(clonedModelData).forEach((model) => {
+    if (model[1].info) {
+      let controllerName = null;
+      if (controllerData) {
+        for (var i = 0; i < controllerData.length; i++) {
+          if (controllerData[i].uuid === model[1].info.controllerUuid) {
```

Comment:
a bit pedantic but can you invert this....you're actually checking if the controller uuid is the same one in the model....so the model goes first 😆

Reply:
Done